lib/src/util/dom_util.dart

```diff
@@ -149,3 +149,27 @@ void setSelectionRange(/* TextInputElement | TextAreaElement */Element input, in
     throw new ArgumentError.value(input, 'input', 'must be an instance of `TextInputElementBase`, `NumberInputElement` or `TextAreaElement`');
   }
 }
+
+/// Custom implementation to avoid issues with `selectionStart` when accessing
+/// an [EmailInputElement] or [NumberInputElement] on Chrome.
+///
+/// If `selectionStart` is not supported on the input, `null` will be returned.
+///
+/// See: <https://bugs.chromium.org/p/chromium/issues/detail?id=324360>
+int getSelectionStart(Element input) {
```

Comment:
The reason I used `Element` here is because, from what I understand, `TextAreaElement` does not inherit from `InputElement`, but we still want to support it in this function. It is the same reason why `setSelectionRange` also accepts an `Element` as its parameter.